File: java/6- Searching/Q3.java
public class Q3 {
    public static void main(String[] args) {

        char[] letters = { 'c', 'f', 'j' };
        char target = 'a';

        int s = 0;
        int e = letters.length - 1;

        while (s <= e) {
            int mid = s + (e - s) / 2;

            if (target < letters[mid]) {
                s = mid + 1;
            } else if (target > letters[mid]) {
                e = mid - 1;
            }

        }
        System.out.println(s % letters.length);
        // return s % letters.length;

    }
}

File: Learning Recursion/problem solving/easy/Q3.py
def search(arr,tar,s,e):
    if s > e:
        return -1
    
    mid = (s + e)//2

    if arr[mid] == tar:
        return mid
    
    if arr[mid] < tar:
       return search(arr,tar,mid+1,e)
    else:
       return search(arr,tar,s,mid-1)
    
nums = [-1,0,3,5,9,12]
target = 2

print(search(nums,target,0,len(nums)-1))
    